nemo_rl/environments/genrm_environment.py

```diff
@@ -0,0 +1,277 @@
+                })
+                
+            else:
+                reward = self.format_penalty
+                try:
+                    if meta["num_responses"] == 1:
+                        if meta["helpfulness_1"] is not None and individual_scores:
+                            reward = -self.distance_abs(individual_scores[0], meta["helpfulness_1"])
+                    
+                    elif meta["num_responses"] == 2:
+                        # Calculate distance for both individual scores
+                        if meta["helpfulness_1"] is not None and individual_scores and len(individual_scores) == 2 and meta["preference_ranking"] is not None and preference_ranking:
+                            reward = - self.distance_abs(individual_scores[0], meta["helpfulness_1"])  - self.distance_abs(individual_scores[1], meta["helpfulness_2"]) - self.distance_abs(preference_ranking, meta["preference_ranking"])
+                
+                except Exception as e:
+                    logging.error(f"Error processing response: {e}")
+
+                rewards.append(float(reward))
+                observations.append({
+                    "role": "environment",
+                    "content": f"Format correct. GenRM evaluation complete. Reward: {reward}"
+                })
+        
+        rewards_tensor = torch.tensor(rewards, dtype=torch.float32)
+        terminateds = torch.ones_like(rewards_tensor, dtype=torch.bool)
+        
+        return EnvironmentReturn(
+            observations=observations,
+            metadata=metadata,
+            next_stop_strings=[None] * len(message_log_batch),
+            rewards=rewards_tensor,
+            terminateds=terminateds,
+        )
+    
+    def global_post_process_and_metrics(self, batch: BatchedDataDict) -> tuple[BatchedDataDict, dict]:
+        """Post processing and metrics calculation."""
+        rewards = batch.get("rewards", torch.tensor([]))
+        num_samples = len(batch.get("idx", []))
+        
+        if len(rewards) == 0:
+            return batch, {}
+        
+        # Convert rewards to numpy for easier computation
+        rewards_np = rewards.numpy() if hasattr(rewards, 'numpy') else np.array(rewards)
+        
+        # Calculate metrics
+        mean_reward = float(np.mean(rewards_np))
+        
+        # Calculate format violation rate (rewards equal to format_penalty)
+        format_violation_rate = float(np.mean(rewards_np == self.format_penalty))
+        
+        # For non-format-penalty rewards, calculate accuracy metrics
+        accuracy_rewards = rewards_np[rewards_np != self.format_penalty]
+        if len(accuracy_rewards) > 0:
+            mean_accuracy_reward = float(np.mean(accuracy_rewards))
+            perfect_pred = float(np.mean(accuracy_rewards == 0))  # Distance 0 = perfect
+            good_pred = float(np.mean(accuracy_rewards >= -10))  # Distance <= 10 = good
```

Comment:
How is that good? With 1 response it'll always be considered as good, and with 2 responses this allows for a 3-point mistake on all three criteria which is pretty large.